secure_enclave/TEUtils.cpp

```diff
@@ -65,152 +70,150 @@ std::string fieldElementToString(const T &field_elem, int base = 10) {
 
     SAFE_CHAR_BUF(arr, BUF_LEN);
 
-    char *tmp = mpz_get_str(arr, base, t);
+    char *hex = mpz_get_str(arr, 16, t);
 
-    ret = std::string(tmp);
+    ret = hex;
 
-    goto clean;
+    int n_zeroes = numBytes * 2 - ret.length();
+    if (n_zeroes > 0) {
+      ret.insert(0, n_zeroes, '0');
+    }
 
   } catch (std::exception &e) {
     LOG_ERROR(e.what());
-    goto clean;
   } catch (...) {
     LOG_ERROR("Unknown throwable");
-    goto clean;
   }
 
-clean:
   mpz_clear(t);
   return ret;
```

Comment:
Here if an error occurs, the exception is caught and potentially invalid value ret is returned.

How does the caller of the function know that an error occured, so it can be handled?

Reply:
In general, in case of an error a function should

1) clean resources so there is no memory leak
2) either return an error as a separate variable or throw an exception


Please check the pull request for this )